### PR TITLE
Add the repository transfer endpoint

### PR DESCRIPTION
**Pull request by m1guelpf**
This PR adds the repo transfer endpoint to the client.
Documentation: https://developer.github.com/v3/repos/#transfer-a-repository

:octocat:

### CODE REVIEW

**Comment by m1guelpf** [comment]
@acrobat Done!

File: lib/Github/Api/Repo.php [hunk 1]
```diff
@@ -659,4 +659,24 @@ public function replaceTopics($username, $repository, array $topics)
...
+     * @link https://developer.github.com/v3/repos/#transfer-a-repository
+     *
+     * @param string $username
+     * @param string $repository
+     * @param string $new_owner
+     * @param array  $team_id
+     *
+     * @return array
+     */
+    public function transfer($username, $repository, $new_owner, $team_id = [])
```
Comment (Nyholm) [inline lib/Github/Api/Repo.php]: We should use camelCase and not snake_case

Comment (Nyholm) [reply lib/Github/Api/Repo.php]: @GrahamCampbell Is there any configuration in StyleCI that would allow us to force camel case?

**Comment by m1guelpf** [comment]
@Nyholm Fixed!

**Comment by m1guelpf** [comment]
@Nyholm @acrobat Status?

File: doc/repos.md [hunk 2]
```diff
@@ -329,3 +329,14 @@ $topics = $client->api('repo')->topics('ornicar', 'php-github-api');
 ```php
 $currentTopics = $client->api('repo')->replaceTopics('ornicar', 'php-github-api', ['new', 'topics']);
 ```
+
+### Transfer a repo to another user
+
+```php
+$currentTopics = $client->api('repo')->transfer('KnpLabs', 'php-github-api', 'github');
```
Comment (acrobat) [inline doc/repos.md]: Use a correct variable name (or remove it if the api does not return anything)

File: doc/repos.md [hunk 3]
```diff
@@ -329,3 +329,14 @@ $topics = $client->api('repo')->topics('ornicar', 'php-github-api');
...
+```
+You can optionally assign some teams by passing an array with their ID's if you're transferring the repo to an organization:
+
+```php
+$currentTopics = $client->api('repo')->transfer('KnpLabs', 'php-github-api', 'github', [1234]);
```
Comment (acrobat) [inline doc/repos.md]: Same as above

**Comment by m1guelpf** [comment]
@acrobat Done!

**Comment by acrobat** [comment]
Thanks @m1guelpf!

**Pull request merged**